pkg/storage/image.go

```diff
@@ -616,6 +625,9 @@ func GetImageService(ctx context.Context, sc *types.SystemContext, store storage
 			return nil, err
 		}
 		for _, r := range systemRegistries {
+			if r.Blocked {
+				continue
+			}
```

Comment:
The other branch of the if (`cleanRegistries` etc.) does not need the Blocked flag?

Reply:
If `ResolveNames` is taught to check the `Blocked` flag for the complete image name created by combining a search registry with an unqualified input, this should be unnecessary.